File: erp-ui/src/app/models/registerRequest.ts
import {UserRole} from "./userRole-model";
import {Poste} from "./poste-model";

export class RegisterRequest {
  id!:number;
  name!:string;
  surname!:string;
  password!:string;
  email!:string;
  naissance!:Date;
  cin!:number;
  departmentId!:string;
  poste!:Poste;
  categorie!:string;
  phoneNumber!:string;
  role!:UserRole;
  adresse!:string
}

File: src/main/java/com/erp/erp/entity/RegisterRequest.java
package com.erp.erp.entity;


import com.erp.erp.enums.Poste;
import com.erp.erp.enums.UserRole;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RegisterRequest  {
    private Long id;
    private String name;
    private String surname;
    private String password;
    private String email;
    private Date naissance;
    private Long cin;
    private String departmentId;
    @Enumerated(EnumType.STRING)
    private Poste poste;
    private String categorie;
    private String phoneNumber;
    @Enumerated(EnumType.STRING)
    private UserRole role;
    private String adresse;

}
